core/python/fileio.py

```diff
@@ -0,0 +1,50 @@
+from spt3g import core
+
+__all__ = ["G3File"]
+
+
+class G3File(object):
+    """Iterable class for G3 files, as created by G3Writer. Loop through frames by doing something like:
+
+        f = core.G3File("/path/to/file.g3")
+        for frame in f:
+            print( frame )
+
+    An entire file can also be read into an indexable list by doing:
+
+        f = list(core.G3File("/path/to/file.g3"))
+    """
+
+    def __init__(self, path):
+        self.reader = core.G3Reader(path)
+
+    def __iter__(self):
+        return self
+
+    def next(self):
+        frames = self.reader.Process(None)
+        if len(frames) == 0:
+            raise StopIteration("No more frames in file")
+        if len(frames) > 1:
+            raise ValueError("Too many frames returned by reader")
+        return frames[0]
+
+    __next__ = next
+
+
+# writer context methods
+
+
+def writer_enter(self):
+    return self
```

Comment:
Should these have initial underscores? They are now getting pulled into the root Python context because of the import * above.

Reply:
I added an `__all__` definition at the top to avoid polluting the namespace.

Reply:
Indeed you did. Sorry I didn't see it!